python/mxnet/model.py

```diff
@@ -113,25 +127,36 @@ def _update_params_on_kvstore(param_arrays, grad_arrays, kvstore, param_names):
         # push gradient, priority is negative index
         kvstore.push(name, grad_list, priority=-index)
         # pull back the weights
-        kvstore.pull(name, arg_list, priority=-index)
+        if _contains_non_default_storage(arg_list):
```

Comment:
Consider arg_list has 1  row_sparse weight and many dense weights, ```_contains_non_default_storage(arg_list)``` will be true, so user have to call  ```kvstore.pull``` for all the dense weights manually out of ```model.py```.  maybe there is some way to split them (row_sparse and dense) so we can only pull row_sparse weights manually?